### PR TITLE
test: [POM] Migrate Send tx Revoke Permissions spec

### LINKED ISSUE #29276: [POM] Migrate Send tx Revoke Permissions spec
### What is this about?

Migrate Send tx Revoke Permissions spec test/e2e/tests/request-queuing/sendTx-revokePermissions.spec.js to POM and ts

### Scenario

_No response_

### Design

_No response_

### Technical Details

_No response_

### Threat Modeling Framework

_No response_

### Acceptance Criteria

_No response_

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### References

_No response_

### PR DESCRIPTION
**Pull request by seaona**
## **Description**

- Migrates `test/e2e/tests/request-queuing/sendTx-revokePermissions.spec.js` spec to POM and ts
- Updates method classes to accommodate new functions needed in this spec
- [A new bug](https://github.com/MetaMask/metamask-extension/issues/29272) has been discovered thanks to this migration, and a new TODO has been added, to add a new spec to cover that flow, once the bug is fixed 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/29273?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/29276

## **Manual testing steps**

1. Check ci
2. Run test ` yarn test:e2e:single test/e2e/tests/request-queuing/sendTx-revokePermissions.spec.ts --browser=chrome --leave-running=true`

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/request-queuing/sendTx-revokePermissions.spec.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import {
+  withFixtures,
+  defaultGanacheOptions,
+  WINDOW_TITLES,
+} from '../../helpers';
+
+describe('Request Queuing', function () {
+  // TODO: add a new spec which checks that after revoking and connecting again
+  // a pending tx is still closed when using revokePermissions.
+  // To be done once this bug is fixed: #29272
```
Comment (seaona) [inline test/e2e/tests/request-queuing/sendTx-revokePermissions.spec.ts:17]: a new bug has been discovered while migrating this spec.
Once the bug is fixed, we should add a new spec to verify that flow too.
Basically the first time we perform the flow it works as expected, but if we re-conenct to the dapp, trigger a new tx, and revoke permissions, that new tx is not cleared.

File: test/e2e/page-objects/pages/test-dapp.ts [hunk 2]
```diff
@@ -321,6 +321,26 @@ class TestDapp {
...
+  /**
+   * Verifies the accounts connected to the test dapp.
+   *
+   * @param connectedAccounts - The expected connected accounts separated by a comma. If no accounts are connected we can omit the param.
+   */
+  async check_connectedAccounts(connectedAccounts: string = '') {
+    console.log('Verify connected accounts');
+    if (connectedAccounts) {
+      await this.driver.waitForSelector({
+        css: '#accounts',
```
Comment (chloeYue) [inline test/e2e/page-objects/pages/test-dapp.ts:333]: could we use the variable defined here? https://github.com/MetaMask/metamask-extension/blob/0e10bab6bc28bb83916c14426fa0439f69903418/test/e2e/page-objects/pages/test-dapp.ts#L35

Comment (seaona) [reply test/e2e/page-objects/pages/test-dapp.ts]: oh absolutely! changed [here](https://github.com/MetaMask/metamask-extension/pull/29273/commits/90dfcf15ac619427f73833ef3d1b7a965978174d), thank you!

**Review by cmd-ob — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM !

**Pull request merged**